src/transformers/models/idefics/configuration_idefics.py

```diff
@@ -0,0 +1,197 @@
+    >>> configuration = IdeficsConfig()
+
+    >>> # Initializing a model from the idefics-9b style configuration
+    >>> model = IdeficsModel(configuration)
+
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+    ```"""
+    model_type = "idefics"
+
+    def __init__(
+        self,
+        vocab_size=32000,
+        additional_vocab_size=0,
+        hidden_size=4096,
+        intermediate_size=11008,
+        num_hidden_layers=32,
+        num_attention_heads=32,
+        dropout=0.0,
+        hidden_act="silu",
+        initializer_range=0.02,
+        alpha_initializer="ones",
+        alphas_initializer_range=0.0,
+        alpha_type="vector",
+        rms_norm_eps=1e-6,
+        use_cache=True,
+        pad_token_id=0,
+        bos_token_id=1,
+        eos_token_id=2,
+        tie_word_embeddings=False,
+        cross_layer_interval=1,
+        qk_layer_norms=False,
+        freeze_text_layers=True,
+        freeze_text_module_exceptions=[],
+        freeze_lm_head=False,
+        freeze_vision_layers=True,
+        freeze_vision_module_exceptions=[],
+        vision_model_name="google/vit-base-patch16-224",
+        vision_embed_dim=768,
+        vision_image_size=224,
+        vision_intermediate_size=5120,
+        vision_patch_size=14,
+        vision_num_hidden_layers=32,
+        vision_num_attention_heads=16,
+        use_resampler=False,
+        resampler_n_latents=64,
+        resampler_depth=6,
+        resampler_n_heads=16,
+        resampler_head_dim=96,
+        **kwargs,
+    ):
+        self.vocab_size = vocab_size
+        self.additional_vocab_size = additional_vocab_size
+        self.hidden_size = hidden_size
+        self.intermediate_size = intermediate_size
+        self.num_hidden_layers = num_hidden_layers
+        self.num_attention_heads = num_attention_heads
+        self.dropout = dropout
+        self.hidden_act = hidden_act
+        self.initializer_range = initializer_range
+        self.alpha_initializer = alpha_initializer
+        self.alphas_initializer_range = alphas_initializer_range
+        self.alpha_type = alpha_type
+        self.rms_norm_eps = rms_norm_eps
+        self.use_cache = use_cache
+
+        self.cross_layer_interval = cross_layer_interval
+        self.qk_layer_norms = qk_layer_norms
+        self.freeze_vision_layers = freeze_vision_layers
+        self.vision_model_name = vision_model_name
+
+        self.freeze_text_layers = freeze_text_layers
+        self.freeze_text_module_exceptions = freeze_text_module_exceptions
+        self.freeze_vision_module_exceptions = freeze_vision_module_exceptions
+        self.freeze_lm_head = freeze_lm_head
+
+        self.vision_embed_dim = vision_embed_dim
+        self.vision_image_size = vision_image_size
+        self.vision_intermediate_size = vision_intermediate_size
+        self.vision_patch_size = vision_patch_size
+
+        self.vision_config_dict = {}
```

Comment:
It might be cleaner to have a subconfig for the vision config (like CLIP does).

Reply:
@younesbelkada - so you were right, let's do it.

Reply:
also please see https://huggingface.slack.com/archives/C02UAKD75L7/p1690428663134799?thread_ts=1690293660.779809&cid=C02UAKD75L7